test/modules/DirectMethodSender/ReporterClients/ReporterClientBase.cs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.ModuleUtil.ReporterClients
+{
+    using System;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil.TestResults;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    public abstract class ReporterClientBase : IDisposable
+    {
+        ILogger logger;
+
+        protected ReporterClientBase(ILogger logger)
+        {
+            this.logger = Preconditions.CheckNotNull(logger, nameof(logger));
+        }
+
+        public abstract void Dispose();
+
+        // TODO: Change this create signature once the TRC and Analyzer URL is merged
+        public static async Task<ReporterClientBase> CreateAsync(
+            ILogger logger,
+            Option<Uri> reportingEndpointUrl,
+            TransportType transportType)
+        {
+            if (reportingEndpointUrl.HasValue)
+            {
+                return new TestResultReporterClient(
+                        reportingEndpointUrl.Expect(() => new ArgumentException("testReportCoordinatorUrl is not expected to be empty")),
```

Comment:
move it in a single line.
```suggestion
                        reportingEndpointUrl.OrDefault()),
```